plugins/env/app/models/environment_variable.rb

```diff
@@ -19,15 +19,14 @@ class << self
     # but return a value with a helpful message
     # also used by an external plugin
     def env(project, deploy_group, preview: false, resolve_secrets: true)
-      variables = nested_variables(project)
-      variables.sort_by!(&:priority)
-      env = variables.each_with_object({}) do |ev, all|
-        all[ev.name] = ev.value if !all[ev.name] && ev.matches_scope?(deploy_group)
+      repo_vars = {}
```

Comment:
use `env = {}` and then `merge!` things into it

Reply:
Yeah ... I thought that about that ...

Reply:
Done